pkg/util/helper/workstatus.go

```diff
@@ -159,3 +159,15 @@ func IsResourceApplied(workStatus *workv1alpha1.WorkStatus) bool {
 	}
 	return false
 }
+
+// IsAppliedContains checking weather work contains resource kind or not.
+func IsAppliedContains(workStatus *workv1alpha1.WorkStatus, kind schema.GroupVersionKind) bool {
```

Comment:
```suggestion
// IsWorkContains checks if the target resource exists in a work.
// Note: This function checks the Work object's status to detect the target resource, so the Work should be 'Applied',
// otherwise always returns false.
func IsWorkContains(workStatus *workv1alpha1.WorkStatus, targetResource schema.GroupVersionKind) bool {
```